src/webgpu/shader/validation/expression/call/builtin/pack2x16float.spec.ts

```diff
@@ -0,0 +1,57 @@
+const kFn = 'pack2x16float';
+export const description = `Validate ${kFn}`;
+
+import { makeTestGroup } from '../../../../../../common/framework/test_group.js';
+import { keysOf } from '../../../../../../common/util/data_tables.js';
+import { kValue } from '../../../../../../webgpu/util/constants.js';
+import { ShaderValidationTest } from '../../../shader_validation_test.js';
+
+const kArgCases = {
+  good: '(vec2f())',
+  bad_0args: '()',
+  bad_2args: '(vec2f(),vec2f())',
+  bad_i32: '(1i)',
+  bad_f32: '(1f)',
+  bad_u32: '(1u)',
+  bad_f32_over_largest_positive_f16_arg0: `(vec2f(f32(${kValue.f16.positive.max}) + 1.0, 0))`,
+  bad_f32_beyond_smallest_negative_f16_arg0: `(vec2f(f32(${kValue.f16.negative.min}) - 1.0, 0))`,
+  bad_f32_over_largest_positive_f16_arg1: `(vec2f(0, f32(${kValue.f16.positive.max}) + 1.0))`,
+  bad_f32_beyond_smallest_negative_f16_arg1: `(vec2f(0, f32(${kValue.f16.negative.min}) - 1.0))`,
+  bad_bool: '(false)',
+  bad_vec4f: '(vec4f())',
+  bad_vec4u: '(vec4u())',
+  bad_vec4i: '(vec4i())',
+  bad_vec4b: '(vec4<bool>())',
+  bad_vec3f: '(vec3f())',
+  bad_array: '(array(1.0, 2.0, 3.0, 4.0))',
+  bad_struct: '(modf(1.1))',
+};
+const kGoodArgs = kArgCases['good'];
+const kReturnType = 'u32';
+
+export const g = makeTestGroup(ShaderValidationTest);
+
+g.test('args')
+  .desc(`Test compilation failure of ${kFn} with various numbers of and types of arguments`)
+  .params(u => u.combine('arg', keysOf(kArgCases)))
+  .fn(t => {
+    t.expectCompileResult(t.params.arg === 'good', `const c = ${kFn}${kArgCases[t.params.arg]};`);
```

Comment:
Would be good to test the pipeline-creation error for OOBs override expression case too.

Reply:
See: https://github.com/gpuweb/cts/blob/7e50fe3c028c3362af01380430472782e12a255f/src/webgpu/shader/validation/expression/call/builtin/const_override_validation.ts#L153

Reply:
Done